projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-types/config-attribute-numeric-input-field/config-attribute-numeric-input-field.component.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  formatNumber,
+  getLocaleNumberSymbol,
+  NumberSymbol,
+} from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { AbstractControl, FormControl, ValidatorFn } from '@angular/forms';
+import { Configurator, LanguageService } from '@spartacus/core';
+import { ConfigFormUpdateEvent } from '../../config-form/config-form.event';
+import { ConfigUIKeyGeneratorService } from '../../service/config-ui-key-generator.service';
+
+@Component({
+  selector: 'cx-config-attribute-numeric-input-field',
+  templateUrl: './config-attribute-numeric-input-field.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ConfigAttributeNumericInputFieldComponent implements OnInit {
+  attributeInputForm: FormControl;
+  numericFormatPattern;
+  locale: string;
+
+  constructor(
+    public uiKeyGenerator: ConfigUIKeyGeneratorService,
+    public languageService: LanguageService
+  ) {}
+
+  @Input() attribute: Configurator.Attribute;
+  @Input() group: string;
+  @Input() ownerKey: string;
+
+  @Output() inputChange = new EventEmitter<ConfigFormUpdateEvent>();
+
+  mustDisplayValidationMessage(): boolean {
+    const wrongFormat: boolean =
+      (this.attributeInputForm.dirty || this.attributeInputForm.touched) &&
+      this.attributeInputForm.errors?.wrongFormat;
+
+    return wrongFormat;
+  }
+
+  determineCurrentLocale() {
+    //locales are available as languages in the commerce backend
+    this.languageService
+      .getActive()
+      .subscribe((language) => (this.locale = language))
+      .unsubscribe();
+  }
+
+  compilePatternForValidationMessage(
+    decimalPlaces: number,
+    totalLength: number
+  ) {
+    let input: string = 10 ** totalLength - 1 + '';
+    if (decimalPlaces > 0) {
+      input =
+        input.substring(0, totalLength - decimalPlaces) +
+        '.' +
+        input.substring(totalLength - decimalPlaces, totalLength);
+    }
+    const inputAsNumber: number = Number(input);
+    this.numericFormatPattern = formatNumber(
+      inputAsNumber,
+      this.locale,
+      '1.' + decimalPlaces + '-' + decimalPlaces
+    ).replace(/9/g, '#');
+  }
+
+  ngOnInit() {
+    this.determineCurrentLocale();
+
+    this.attributeInputForm = new FormControl('', [
+      this.numberFormatValidator(
+        this.locale,
+        this.attribute.numDecimalPlaces,
+        this.attribute.numTotalLength
+      ),
+    ]);
+
+    const numDecimalPlaces = this.attribute.numDecimalPlaces;
+
+    this.compilePatternForValidationMessage(
+      numDecimalPlaces,
+      this.attribute.numTotalLength
+    );
+
+    if (this.attribute.userInput) {
+      this.attributeInputForm.setValue(
+        formatNumber(
+          Number(this.attribute.userInput),
+
+          this.locale,
```

Comment:
What will happen when `this.locale` (active lang) was not registered using angular `registerLocaleData()`? Will it throw an error? If yes, maybe we should fallback to `en`?

Reply:
Yes, agree. Implemented this fallback like in the date pipe.